### ISSUE TITLE
Restructuring of the database

### ISSUE THREAD

**Issue by lindell** [phixofor/unvis.it#6]
The articles are saved as the html that should be shown on the unvis.it website. I assume this was made because it is the easiest way of doing it when the page was made. This should really be changed to something where only the data i saved. It will require some restructuring of the database with multiple tables etc. 

I could/would like to do this but want to ensure that it will be used in that case since this will require a reset of the db if someone is not writing a script to transfer the old format to the new (which I do not plan to do).

**Comment by phixofor** [comment]
Yes it was the easiest way, before that they were saved as plain textfiles. 

In the current deploy the database resets every 30 days to keep the articles "fresh", in case they are updated or so. Most pages only have around 5-6 days of activity, often with a peak day 1 or 2. 

I would definitely welcome a change to this, if it brings optimization and stability. Since everything has been pretty lo-fi from the start, very few things have screwed up a long the way – and as I am not a qualified developer I value that stability. 

How exactly do you propose the "data" would be saved instead of just the result from the phpReadability script?

**Comment by lindell** [comment]
The thing that is saved to the db is not only the result of the phpReadability script but the entire bootstrap div that should be shown.

The simplest thing would be to store the title and body separate and not include any html that belongs to the unvis.it. This way the structure of the website may be changed at any time and is not dependent upon the way things are saved in the db.

Then it would be nice to save things such as references in a separate table, but after looking at the readability script it does not seem like it have the functionality so that will probably not be done.

**Comment by lindell** [comment]
I have now separated the title and body and do no longer include any unnecessary unvis.it html.
You expressed your dislike to with keeping things in different files and I hope this will change your mind.

Functions of the site is now separated into parts where you can actually see whats happening instead of one big index.php.

I made two new directories. One is for views and one is for controllers. The views is only allowed to have code that displays data and should not handle it in any other way while the controllers can only handle the data but not display it. Something similair to [MVC](https://en.wikipedia.org/wiki/Model%E2%80%93view%E2%80%93controller) but with maybe an not so great model structure.

TL;DR: Less spaghetticode, more structure.

This is not yet pull ready but at least it's a sneak peak.

The code:
https://github.com/lindell/unvis.it/tree/big-cleanup

The new db:
```
CREATE TABLE IF NOT EXISTS `cached` (
  `id` int(11) NOT NULL AUTO_INCREMENT,
  `url` text CHARACTER SET latin1 NOT NULL,
  `title` text CHARACTER SET latin1 NOT NULL,
  `body` text CHARACTER SET latin1 NOT NULL,
  `time` timestamp NOT NULL DEFAULT CURRENT_TIMESTAMP,
  PRIMARY KEY (id)
) DEFAULT CHARSET=utf8;
```
